### DISCUSSION TITLE
Multiple page app using beeware.

### DISCUSSION THREAD

**Discussion by ahmed-881994** [beeware/beeware#329] [Q&A]
I'm researching python cross platform framework for mobile development and beeware stood out and i'm really considering adopting it .However, I can't find a standard way to create multipage app. Is there any documents or tutorials describing how to do it? I know there isn't a discussion describing what I need.

**Comment by freakboy3742** [comment]
It depends a little on what you mean by "multipage app". 

The simplest interpretation is no more than replacing the content of the main window with new content. This can be done with exactly that API: assigning the `content` attribute of a window will cause the content of the window to be replaced. Every Toga example includes assigning `self.main_window.content = <some box>`; if you maintain multiple `<some box>` objects, you can switch between them as your UI requires.

There is also the `OptionContainer` widget - that adds a tab control to switch between multiple pages of content. The Toga repository contains a collection of example projects, including an `optioncontainer` example that demonstrates this.

If you're looking for a more complex pre-built "navigation" between pages of content, such as the classic UI on a mobile app where clicking on a link causes a new page of content to "slide in" from the right, with a "back" button in the titlebar to go back - Toga doesn't currently have an API for this, but it is on our TODO list.

**Reply by ahmed-881994** [reply]
I figured that out but the question is will I then treat the boxes as objects or will they be the return of methods in the main class?
Is there like a standard for this operation? (can you give an example on how to manage it?) Especially that I intend to have a SignIn page that when authenticated it takes the user to an OptionContainer and it won't be maintainable to write the entire app as one class.

**Reply by freakboy3742** [reply]
The questions you're asking are more about "large project code management" than anything I'm in a position to provide a canonical answer for. From a technical perspective, there are no requirements on how to manage window content. If you want to construct everything in the `startup()` method, you can; if you want to have standalone methods or classes to manage construction of individual "pages" of content, you can do that too. The only complication is that it's up to you to pass around relevant context (such as the app or window instance), but there's nothing "special" about any of that - it's standard variable management stuff.
